gamechange.cpp

```diff
@@ -12,8 +12,11 @@ void GameChange::apply(std::shared_ptr<GameChange> self)
         old_assoziation = dot;
         dot->removeField(affected_field);
     }
-    affected_field->assigned_dot = new_assoziation;
-    new_assoziation->registerField(affected_field);
+    if (new_assoziation)
+    {
+        affected_field->assigned_dot = new_assoziation;
+        new_assoziation->registerField(affected_field);
+    }
```

Comment:
I don't really get this part tbh, this whole create pointer and pass self as param and stuff...
But I'm a noob in C++ so... "Wird scho bassen!"